File: src/main/java/cc/mirukuneko/realtrainmodrenewed/rail/util/RailDir.java
package cc.mirukuneko.realtrainmodrenewed.rail.util;

public enum RailDir {
    RIGHT(-1),
    LEFT(1),
    NONE(0);

    public final byte id;

    RailDir(int id) {
        this.id = (byte) id;
    }

    public RailDir invert() {
        return this == RIGHT ? LEFT : this == LEFT ? RIGHT : NONE;
    }
}

File: src/main/kotlin/cc/mirukuneko/realtrainmodrenewed/rail/util/RailDir.kt
package cc.mirukuneko.realtrainmodrenewed.rail.util

enum class RailDir(val id: Byte) {
    RIGHT(-1),
    LEFT(1),
    NONE(0);

    fun invert(): RailDir = when (this) {
        RIGHT -> LEFT
        LEFT -> RIGHT
        NONE -> NONE
    }
}
